### PR TITLE
Crs 1835

### PR DESCRIPTION
**Pull request by dave-virden**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/config/FeatureToggles.kt
Comment (srowlands-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/config/FeatureToggles.kt:1]: Is this file necessary?

**Comment by srowlands-moj** [comment]
Are you doing AC 3 in another PR? So the calc of active TUSED + 14 days?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/config/FeatureToggles.kt
Comment (dave-virden) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/config/FeatureToggles.kt:1]: I think so, unless there's a neater way of doing it?

**Comment by dave-virden** [comment]
> Are you doing AC 3 in another PR? So the calc of active TUSED + 14 days?

AC3 (of CRS-1835) doesn't return dates, it's handled by the ValidationService

**Comment by srowlands-moj** [comment]
Apologies 

> > Are you doing AC 3 in another PR? So the calc of active TUSED + 14 days?
> 
> AC3 (of CRS-1835) doesn't return dates, it's handled by the ValidationService

Apologies what I was talking about is in CRS-1852, not sure why I thought it was AC3

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/validation/ValidationService.kt [hunk 1]
```diff
@@ -648,7 +665,12 @@ class ValidationService(
   private fun validateSupportedSentences(sentencesAndOffences: List<SentenceAndOffences>): List<ValidationMessage> {
     val supportedCategories = listOf("2003", "2020")
     val validationMessages = sentencesAndOffences.filter {
-      !(SentenceCalculationType.isSupported(it.sentenceCalculationType) && supportedCategories.contains(it.sentenceCategory))
+      if (SentenceCalculationType.isSupported(it.sentenceCalculationType) && supportedCategories.contains(it.sentenceCategory)) {
+        val type = from(it.sentenceCalculationType)
+        !featureToggles.botus && type.sentenceClazz == BotusSentence::class.java
+      } else {
+        true
+      }
```
Comment (srowlands-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/validation/ValidationService.kt:673]: Rather than doing this, can you not have `isSupported` set from the Feature toggle in `SentenceCalculationType.kt`?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/config/FeatureToggles.kt
Comment (srowlands-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/config/FeatureToggles.kt:1]: Actually I think encapsulating like this is a bit nicer, I'm not sure if i'd have preferred the default to come from the props file though - but that's minor.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/validation/ValidationServiceTest.kt [hunk 2]
```diff
@@ -686,6 +688,77 @@ class ValidationServiceTest {
     )
   }
 
+  @Test
+  fun `Test A Botus sentence consecutive to unsupported`() {
```
Comment (srowlands-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/validation/ValidationServiceTest.kt:692]: Missing `another sentenceType is`?

**Comment by srowlands-moj** [comment]
```***************************
APPLICATION FAILED TO START
***************************

Description:

Failed to bind properties under 'feature-toggles.botus' to boolean:

    Property: feature-toggles.botus
    Value: "${BOTUS_FEATURE_TOGGLE}"
    Origin: class path resource [application.yml] - 130:10
    Reason: failed to convert java.lang.String to boolean (caused by java.lang.IllegalArgumentException: Invalid boolean value '${BOTUS_FEATURE_TOGGLE}')```

You may need to either provide the env var, OR set a fallback for spring in the properties file `:false`

**Review by srowlands-moj — APPROVED** [review]
LGTM
